### PR TITLE
Added export of private data

### PR DESCRIPTION
**Pull request by mika76**
Also some minor changes and fixes

- Changed the username to be a mandatory argument instead of an option (it didn't work if username wasn't input anyway)
- Added --password -p flag - if this is entered it tries to log into delicious, get it's cookies and attach them to the following requests
- Added a private flag to the export json items
- Changed the default output file to be delicious.json instead of results.json - it's just cleaner and more explicit
- Fixed up the npm\yarn build to be cross platform
- Updated readme

### CODE REVIEW

**Comment by mika76** [comment]
This is all linked to #1

**Comment by mika76** [comment]
It seems travis is having a problem downloading a dependency (commander)

**Comment by knalli** [comment]
Thank you. 👍  Maybe splitting up such requests is better. ;)

**Comment by knalli** [comment]
Well, I'm not familiar with yarn yet. (I know what it is, but not in use atm). Well, maybe travis caching issue? I have not enabled any cache there.

**Comment by mika76** [comment]
I will remove the yarn file and it should just use npm - let's see if that solves it...

**Comment by knalli** [comment]
Don't get me wrong: We can add yarn integration, but maybe separated. I'm not against it.

**Comment by mika76** [comment]
Well it shouldn't make a difference really - npm or yarn both usually work without hassle together. Anyway, let's see if this solves it for now.

**Comment by knalli** [comment]
Well, seems to be an issue with master also.

**Comment by mika76** [comment]
Hmm npm is even worse 😆 

Weird. Only new dependencies are node-build-tools and set-cookie-parser and they both worked fine on my machine...

**Comment by knalli** [comment]
Nah, master is failing exactly like here.

File: src/index.js [hunk 1]
```diff
@@ -6,12 +6,16 @@ import program from "commander";
 import pck from "./../package.json";
 
 program.version(pck.version)
-       .option('-u --username [username]', 'Your del.icio.us username')
```
Comment (knalli) [inline src/index.js:9]: Oops, it must be `'-u --username username'` -- brackets defining its being optional. Initially missed by me.

Comment (mika76) [reply src/index.js:9]: Although I changed the username to an argument in my branch so I don't think that's what's causing it...

**Comment by mika76** [comment]
You're not happy with the pull request? As far as I can see nothing I did has made Travis fail, and my testing of the changes is quite positive...

**Comment by knalli** [comment]
Sorry, I have got not found some spare time for this. Will be.

**Comment by mika76** [comment]
ping 🏓

**Comment by knalli** [comment]
Thank you very much @mika76 :)

Landed as bb08937 in the just available 1.1.0

**Comment by mika76** [comment]
Brilliant 😊  👍

**Pull request closed without merging**